examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,123 @@
+"""
+===============================================================================
+Silhouette analysis for sample data clustered using KMeans clustering algorithm
```

Comment:
Let's make this a bit more instructive, like: "Selecting the number of clusters with silhouette analysis"